### PR TITLE
StAX native support

### LINKED ISSUE #1696: StAX native support

### PR DESCRIPTION
**Pull request by jamesnetherton**
Fixes #1696

### CODE REVIEW

File: integration-tests/stax/src/main/java/org/apache/camel/quarkus/component/stax/it/StaxResource.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    ProducerTemplate producerTemplate;
+
+    @Path("/records")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public int staxProcessRecords() throws Exception {
+        InputStream resource = StaxResource.class.getResourceAsStream("/records.xml");
+        TagCountHandler countingHandler = producerTemplate.requestBody("direct:processRecords", resource,
+                TagCountHandler.class);
+        return countingHandler.getCount();
```
Comment (ppalaga) [inline integration-tests/stax/src/main/java/org/apache/camel/quarkus/component/stax/it/StaxResource.java:49]: Is there any reason not to `try(InputStream resource = StaxResource.class.getResourceAsStream("/records.xml"))` here and elsewhere in this class?

**Review by ppalaga — APPROVED** [review]
Thanks, looks great! A minor nitpick inline.

File: integration-tests/stax/src/main/java/org/apache/camel/quarkus/component/stax/it/StaxResource.java (hunk 1 above)
Comment (jamesnetherton) [reply integration-tests/stax/src/main/java/org/apache/camel/quarkus/component/stax/it/StaxResource.java:49]: Good catch - I'll change that.

**Review by oscerd — APPROVED** [review]

File: extensions/stax/deployment/src/main/java/org/apache/camel/quarkus/component/stax/deployment/StaxProcessor.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    @BuildStep
+    ReflectiveClassBuildItem registerForReflection(CombinedIndexBuildItem combinedIndex) {
+        // Register any ContentHandler impls for reflection so they can be used by the sax component
+        IndexView index = combinedIndex.getIndex();
+        String[] contentHandlers = index.getAllKnownImplementors(DotName.createSimple(ContentHandler.class.getName()))
+                .stream()
+                .map(ClassInfo::name)
+                .map(DotName::toString)
+                .peek(System.out::println)
```
Comment (aldettinger) [inline extensions/stax/deployment/src/main/java/org/apache/camel/quarkus/component/stax/deployment/StaxProcessor.java:53]: Should we remove ? or use a logger ?

Comment (jamesnetherton) [reply extensions/stax/deployment/src/main/java/org/apache/camel/quarkus/component/stax/deployment/StaxProcessor.java:53]: Yes, I noticed that debugging leftover. I'll remove it in a subsequent commit.

File: integration-tests/stax/src/main/java/org/apache/camel/quarkus/component/stax/it/StaxResource.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            return countingHandler.getCount();
+        }
+    }
+
+    @Path("/records/split")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public void staxSplitRecords() throws Exception {
+        try (InputStream resource = StaxResource.class.getResourceAsStream("/records.xml")) {
+            MockEndpoint mockEndpoint = context.getEndpoint("mock:split", MockEndpoint.class);
```
Comment (aldettinger) [inline integration-tests/stax/src/main/java/org/apache/camel/quarkus/component/stax/it/StaxResource.java:69]: Interesting idea :+1:

**Review by aldettinger — APPROVED** [review]
Ok, no show stopper then. Looks good to me.

**Pull request merged**